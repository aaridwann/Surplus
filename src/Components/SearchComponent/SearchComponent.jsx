import React from 'react';
import { Text, TextInput, TouchableOpacity, View } from 'react-native';
import { Octicons } from '@expo/vector-icons';
import { scaleSize } from '../../Utils/Scale/Scale';
import PropTypes from 'prop-types';
import styles from './Styles';

const _renderButton = (onPress) => (
	<TouchableOpacity testID='search-button' onPress={onPress}>
		<Text>
			<Octicons name="search" size={scaleSize(20)} color="black" />;
		</Text>
	</TouchableOpacity>
);

const _renderInput = (onChange) => (
	<TextInput testID='search-input' onChangeText={onChange} style={styles.input} />
);

function SearchComponent({ onChange, onPress }) {
	return (
		<View testID='search-container' style={styles.container}>
			{_renderInput(onChange)}
			{_renderButton(onPress)}
		</View>
	);
}

export default SearchComponent;

SearchComponent.propTypes = {
	onChange: PropTypes.func,
	onPress: PropTypes.func,
};
