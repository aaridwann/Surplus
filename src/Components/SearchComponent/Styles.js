import { StyleSheet } from 'react-native';
import { scaleHeight, scaleWidth } from '../../Utils/Scale/Scale';
import Constants from '../../Utils/Constants/Constants';

const styles = StyleSheet.create({
	container: {
		width: scaleWidth(320),
		height: scaleHeight(40),
		backgroundColor: 'white',
		flexDirection: 'row',
		alignItems: 'center',
		top: scaleHeight(-25),
		borderRadius: 10,
		justifyContent: 'space-evenly',
		shadowColor: '#000',
		shadowOffset: {
			width: 0,
			height: 5,
		},
		shadowOpacity: 0.34,
		shadowRadius: 6.27,
		elevation: 10,
	},
	input: {
		textAlign: 'center',
		width: '60%',
		borderRightWidth: 1.5,
		borderColor: Constants.Colors.LightGreenSurplus,
		padding: 4,
		color: 'gray',
		marginHorizontal: 8,
	},
});

export default styles;
