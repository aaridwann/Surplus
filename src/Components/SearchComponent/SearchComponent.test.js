import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import SearchComponent from './SearchComponent';

describe('SearchComponent', () => {
	it('snapshot', () => {
		expect(
			render(<SearchComponent onPress={jest.fn()} onChange={jest.fn()} />)
		).toMatchSnapshot();
	});

	it('renders correctly', () => {
		const { getByTestId } = render(<SearchComponent />);
		const container = getByTestId('search-container');
		expect(container).toBeTruthy();
	});

	it('triggers onChange event', () => {
		const handleChange = jest.fn();
		const { getByTestId } = render(<SearchComponent onChange={handleChange} />);
		const input = getByTestId('search-input');
		fireEvent.changeText(input, 'example');
		expect(handleChange).toHaveBeenCalledWith('example');
	});

	it('triggers onPress event', () => {
		const handlePress = jest.fn();
		const { getByTestId } = render(<SearchComponent onPress={handlePress} />);
		const button = getByTestId('search-button');
		fireEvent.press(button);
		expect(handlePress).toHaveBeenCalled();
	});
});
