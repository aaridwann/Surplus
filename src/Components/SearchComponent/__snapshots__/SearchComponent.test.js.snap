// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 10,
      "flexDirection": "row",
      "height": 80,
      "justifyContent": "space-evenly",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 5,
        "width": 0,
      },
      "shadowOpacity": 0.34,
      "shadowRadius": 6.27,
      "top": -50,
      "width": 640,
    }
  }
  testID="search-container"
>
  <TextInput
    autoCapitalize={false}
    onChangeText={[MockFunction]}
    placeholder="Find someone"
    style={
      Object {
        "borderColor": "#50b8a8",
        "borderRightWidth": 1.5,
        "color": "gray",
        "marginHorizontal": 8,
        "padding": 4,
        "textAlign": "center",
        "width": "60%",
      }
    }
    testID="search-input"
  />
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="search-button"
  >
    <Text>
      ;
    </Text>
  </View>
</View>
`;
